docs/source/common/production_inference.rst

```diff
@@ -1,48 +1,246 @@
+you can use to deploy and use your models for inference.
 
-Exporting to ONNX
------------------
-PyTorch Lightning provides a handy function to quickly export your model to ONNX format, which allows the model to be independent of PyTorch and run on an ONNX Runtime.
+------------
 
-To export your model to ONNX format call the ``to_onnx`` function on your Lightning Module with the filepath and input_sample.
+******************
+With Lightning API
+******************
+
+Following are some possible ways you can use Lightning to run inference in production. Note that this can introduce extra required Lightning dependencies
+in your production environment.
+
+------------
+
+Prediction API
+==============
+
+Lightning provides you with a prediction API that can be accessed using :meth:`~pytorch_lightning.trainer.trainer.Trainer.predict`.
+To configure this with LightningModule, you need to override the :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` method.
+By default :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` calls the :meth:`~pytorch_lightning.core.lightning.LightningModule.forward`
+method. In order to customize this behaviour, simply override the :meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` method.
+
+For the example let's override ``predict_step`` and try out `Monte Carlo Dropout <https://arxiv.org/pdf/1506.02142.pdf>`_:
 
 .. code-block:: python
 
-    filepath = "model.onnx"
+    class LitMCdropoutModel(pl.LightningModule):
+        def __init__(self, model, mc_iteration):
+            super().__init__()
+            self.model = model
+            self.dropout = nn.Dropout()
+            self.mc_iteration = mc_iteration
+
+        def predict_step(self, batch, batch_idx):
+            # enable Monte Carlo Dropout
+            self.dropout.train()
+
+            # take average of `self.mc_iteration` iterations
+            pred = [self.dropout(self.model(x)).unsqueeze(0) for _ in range(self.mc_iteration)]
+            pred = torch.vstack(pred).mean(dim=0)
+            return pred
+
+------------
+
+PyTorch Runtime
+===============
+
+You can also load the saved checkpoint and use it as a regular :class:`torch.nn.Module`.
+
+.. code-block:: python
+
+    class SimpleModel(LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.l1 = torch.nn.Linear(in_features=64, out_features=4)
+
+        def forward(self, x):
+            return torch.relu(self.l1(x.view(x.size(0), -1)))
+
+
+    # create the model
+    model = SimpleModel()
+
+    # train it
+    trainer = Trainer(gpus=2)
+    trainer.fit(model, train_dataloader, val_dataloader)
+    trainer.save_checkpoint("best_model.ckpt", weights_only=True)
+
+    # use model after training or load weights and drop into the production system
+    model = SimpleModel.load_from_checkpoint("best_model.ckpt")
```

Comment:
quick question. Can It be possible to load the jitted model in the LightningModule?

Reply:
yes but not with `load_from_checkpoint`.
```py
model = LitModel(...)
traced = model.to_torchscript(method='script')
model.load_state_dict(traced.state_dict())
```
should we extend the API to load jit checkpoints?

Reply:
Yes, I think it would be great.

Reply:
```py
    class SingleLayerMLP(nn.Module):

        def __init__(self, in_features=64, out_features=4, **kwargs):
            super().__init__(self)
            self.l1 = torch.nn.Linear(in_features=64, out_features=4)

        def forward(self, x):
            return torch.relu(self.l1(x.view(x.size(0), -1)))

    class SimpleModel(LightningModule):
        def __init__(self, in_features=64, out_features=4):
            super().__init__()
            self.mlp = SingleLayerMLP(in_features=in_features, out_features=out_features)

        def forward(self, x):
            return self.mlp(x)


    model = SimpleModel()
    # train it
    trainer = Trainer(gpus=2)
    trainer.fit(model, train_dataloader, val_dataloader)
    trainer.save_checkpoint("best_model.ckpt", weights_only=True)
    # use model after training or load weights and drop into the production system
 
    # In a production environment without PyTorch Lightning.
    checkpoint = torch.load("best_model.ckpt")
    model = SingleLayerMLP(**checkpoint['hyper_parameters'])
    # Remove the MLP key used within the 
    state_dict = {'.'.join(x.split('.')[1:]): v for k in checkpoint['state_dict'].items()}
    model.load_state_dict(state_dict)
```

Reply:
this is `within Lightning` section, using it with native PT is defined below.